scipy/optimize/_numdiff.py

```diff
@@ -570,3 +638,48 @@ def check_derivative(fun, jac, x0, bounds=(-np.inf, np.inf), args=(),
                                    args=args, kwargs=kwargs)
         abs_err = np.abs(J_to_test - J_diff)
         return np.max(abs_err / np.maximum(1, np.abs(J_diff)))
+
+
+class FiniteDifference:
+    """Represents finite difference approximation.
+
+    Provide an easy way of calling `approx_derivative` repeatedly
+    with the same set of parameters. Avoid duplicated calls to ``fun``
+    if ``avoid_duplicated_calls=True``.
```

Comment:
I guess reference `approx_derivative` for other arguments and general info.

(Considering this module is likely to go public).